packages/backend/src/managers/llama-stack/llamaStackManager.ts

```diff
@@ -0,0 +1,251 @@
+import path from 'node:path';
+import fs from 'node:fs/promises';
+import type { ConfigurationRegistry } from '../../registries/ConfigurationRegistry';
+import { getFreeRandomPort } from '../../utils/ports';
+
+export const LLAMA_STACK_CONTAINER_LABEL = 'ai-lab-llama-stack-container';
+export const LLAMA_STACK_API_PORT_LABEL = 'ai-lab-llama-stack-api-port';
+
+export class LlamaStackManager {
+  #initialized: boolean;
+  #containerInfo: LlamaStackContainerInfo | undefined;
+
+  constructor(
+    private readonly appUserDirectory: string,
+    private taskRegistry: TaskRegistry,
+    private podmanConnection: PodmanConnection,
+    private containerRegistry: ContainerRegistry,
+    private configurationRegistry: ConfigurationRegistry,
+    private telemetryLogger: TelemetryLogger,
+  ) {
+    this.#initialized = false;
+    this.podmanConnection.onPodmanConnectionEvent(this.watchMachineEvent.bind(this));
+    this.containerRegistry.onStartContainerEvent(this.onStartContainerEvent.bind(this));
+    this.containerRegistry.onStopContainerEvent(this.onStopContainerEvent.bind(this));
+  }
+
+  private async refreshLlamaStackContainer(id?: string): Promise<void> {
+    const containers = await containerEngine.listContainers();
+    const containerInfo = containers
+      .filter(c => !id || c.Id === id)
+      .filter(c => c.State === 'running' && c.Labels && LLAMA_STACK_CONTAINER_LABEL in c.Labels)
+      .at(0);
+    if ((id && containerInfo) || !id) {
+      this.#containerInfo = containerInfo
+        ? {
+            containerId: containerInfo.Id,
+            port: parseInt(containerInfo.Labels[LLAMA_STACK_API_PORT_LABEL]),
+          }
+        : undefined;
+    }
+  }
+
+  private async watchMachineEvent(event: PodmanConnectionEvent): Promise<void> {
+    if ((event.status === 'started' && !this.#containerInfo) || (event.status === 'stopped' && this.#containerInfo)) {
+      await this.refreshLlamaStackContainer();
+    }
+  }
+
+  private async onStartContainerEvent(event: ContainerEvent): Promise<void> {
+    await this.refreshLlamaStackContainer(event.id);
+  }
+
+  private onStopContainerEvent(event: ContainerEvent): void {
+    console.log('event id:', event.id, ' containerId: ', this.#containerInfo?.containerId);
+    if (this.#containerInfo?.containerId === event.id) {
+      this.#containerInfo = undefined;
+      this.taskRegistry.deleteByLabels({ trackingId: LLAMA_STACK_CONTAINER_TRACKINGID });
+    }
+  }
+
+  async getLlamaStackContainer(): Promise<LlamaStackContainerInfo | undefined> {
+    if (!this.#initialized) {
+      const containers = await containerEngine.listContainers();
+      this.#containerInfo = containers
+        .filter(c => c.State === 'running' && c.Labels && LLAMA_STACK_CONTAINER_LABEL in c.Labels)
+        .map(c => ({
+          containerId: c.Id,
+          port: parseInt(c.Labels[LLAMA_STACK_API_PORT_LABEL]),
+        }))
+        .at(0);
+      this.#initialized = true;
```

Comment:
feels weird to have a `get*` method setting `initialized` probably need to be renamed to something like `init` or `setup`, or `config` etc.

Moreover, I have some issue understanding the logic, could you add some comments to the function? thanks

Last: `get*` method should not return possibly undefined, feel more natural to use find

Reply:
I added some comment. 

I'm trying to be consistent with `instructlabManager`. 

For me, setting initialized if for caching the result, seems legit in a `get` method